src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs

```diff
@@ -298,7 +365,17 @@ public static bool TryParse(string value, out SemanticVersion result)
         /// </returns>
         public static bool operator <(SemanticVersion a, SemanticVersion b)
         {
-            return b > a;
+            if (object.ReferenceEquals(a, null))
+            {
+                if (object.ReferenceEquals(b, null))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            return a.CompareTo(b) < 0;
```

Comment:
This change is not needed. Original implementation is correct:

    return b > a;

Reply:
This change is not needed. Original implementation is correct.

There is no point in changing something that is correct.

It is also about reusing the logic that already exists in the `>` implementation.

Reply:
Done.